### ISSUE TITLE
`TypeChain` is becoming deprecated

### ISSUE THREAD

**Issue by NouDaimon** [solidstate-network/solidstate-solidity#231]
As per https://github.com/dethcrypto/TypeChain/commit/4c9773403a2aa66bf6f5fa066e8ef1b00769c71b it is prudent to consider alternatives to `TypeChain` for testing the `SolidState` contract suite.

Perhaps even considering a switch to Foundry, which would deprecate the need for a `TypeChain` alternative.

**Comment by ItsNickBarry** [comment]
> deprecated but still works perfectly

Yes, it still works well enough that I haven't bothered to replace it, but it has prevented progress in some areas (for example, I tried to use the `hardhat-exposed` plugin for something but it's incompatible: https://github.com/frangio/hardhat-exposed/issues/9).  I'll keep this issue open.
